### DISCUSSION TITLE
[SUGGESTION] multi-GPU (decoding) workload management

### DISCUSSION THREAD

**Discussion by MarcoRavich** [Anime4000/IFME#245] [Ideas]
Hi there, since the development is going ahead recently, I hope you evaluate what is being [discussed here](https://github.com/cdgriffith/FastFlix/discussions/558#discussioncomment-10072409) for IFME too:

> it would be extremely interesting (and maybe not too difficult to implement) to allow user choose which GPU allocate for the decoding workload regardless of which has chosed for the encoding!

Hope that inspires !

(note: activating the Discussion section in GH is also interesting to avoid "issues" - which aren't - like this)

**Comment by Anime4000** [comment]
Hi,

IFME relying decoding work done by FFmpeg, I believe it should be automatically decide

> (note: activating the Discussion section in GH is also interesting to avoid "issues" - which aren't - like this)

I going to activate it

**Comment by MarcoRavich** [comment]
Thanks for reply.

If selected, FFMPEG uses the same GPU for both encoding and decoding, AFAIK.

The idea is to decode with one and encode with another (or with CPU).

Some resources that *may* inspire:
- https://sourceforge.net/p/ffmpeg-batch/discussion/general/thread/d1aef6fc96/ (bumped)
- https://arstech.net/ffmpeg-gpu-transcoding-examples/
- https://medium.com/gumgum-tech/hardware-accelerated-transcoding-a-marriage-between-ffmpeg-containers-and-nvidia-cd8624d5ed00
- https://github.com/Intel-Media-SDK/MediaSDK/wiki/FFmpeg-QSV-Multi-GPU-Selection-on-Linux
- https://www.reddit.com/r/ffmpeg/comments/fr03mg/hq_ffmpeg_encoding_with_gpu_nvenc_everything_ive/

Thanks in advance.
